app/interactors/tasks/destroy/send_notification.rb

```diff
@@ -0,0 +1,29 @@
+module Tasks
+  class Destroy
+    class SendNotification
+      include Interactor
+
+      delegate :project, :task, :user, to: :context
+
+      def call
+        mail_owner.deliver_later unless ProjectMembership.find_by(project: project, user: user).owner?
```

Comment:
If I am a project member, will I receive 1 email or 2?

Reply:
No, if you were project member, this part of code would send email only to project owner:

```
def mail_owner
  @mail_owner ||= TaskMailer.task_destroyed_to_owner(project, task)
end
...
class TaskMailer < ApplicationMailer
  def task_destroyed_to_owner(project, task)
    mail(to: project.users.where(project_memberships: { role: :owner }).pluck(:email))
  end
end
```

And mail_members would send mail to every member of project except owner and you(initiator):
```
def mail_members
  @mail_members ||= TaskMailer.task_destroyed_to_members(project, task, user)
end
...
class TaskMailer < ApplicationMailer
  def task_destroyed_to_members(project, task, user)
    recipients = project.users.where(project_memberships: { role: :member }).where.not(id: user).pluck(:email)
  
    mail(to: recipients)
  end
end
```